assembly/BytesReader.ts

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020 LimeChain Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Bool } from "./Bool";
+import { UInt8, UInt16, UInt32, UInt64 } from "./UInt";
+import { Int8, Int16, Int32, Int64, CompactInt } from './Int';
+import { Hash } from './Hash';
+import { ByteArray } from "./Arrays";
+import { ScaleString } from "./ScaleString";
+
+/**
+ * ByteReader class that helps reading bytes into Scale-Codec types
+ */
+export class BytesReader{
```

Comment:
I think that we should add `@description` tags for the inline docs so that they are shown in editors